crates/sui-types/src/temporary_store.rs

```diff
@@ -214,9 +218,18 @@ impl<'backing> TemporaryStore<'backing> {
             protocol_config: protocol_config.clone(),
             loaded_child_objects: BTreeMap::new(),
             runtime_packages_loaded_from_db: RwLock::new(BTreeMap::new()),
+            formerly_deleted_shared_objects: BTreeMap::new(),
         }
     }
 
+    /// Add any found deleted shared object to the Temporary Store
+    pub fn update_formerly_deleted_shared_objects(
+        &mut self,
+        deleted_shared_objects: DeletedSharedObjects,
+    ) {
+        self.formerly_deleted_shared_objects = deleted_shared_objects;
```

Comment:
I see this field being assigned, but where is it accessed/read?

Reply:
Now that you mention this, I see that we don't need it after all, which is great news because it simplifies things nicely.